sklearn/linear_model/_ridge.py

```diff
@@ -288,11 +288,14 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None, copy=False):
 def _solve_svd(X, y, alpha, xp=None):
     xp, _ = get_namespace(X, xp=xp)
     U, s, Vt = xp.linalg.svd(X, full_matrices=False)
-    idx = s > 1e-15  # same default value as scipy.linalg.pinv
-    s_nnz = s[idx][:, None]
+    idx = s > 1e-15[:, None]  # same default value as scipy.linalg.pinv
```

Comment:
`>` does not have precedence of over `[]`, right? Trying this pattern locally yields:

```python
TypeError: 'float' object is not subscriptable
```

I think you meant the following instead:

```suggestion
    # scipy.linalg.pinv also thresholds at 1e-15 by default.
    idx = (s > 1e-15)[:, None]
```

Reply:
But then the following `s = s[:, None]` seems redundant and similarly for `idx[:, None]` in the call to `where`.

Also we should probably rename `idx` to something more correct and explicit such as `strictly_positive_mask`.

Reply:
Thanks! I think I had this working locally and passing tests, but I must have messed something up on the merge.

I'm planning on circling back to this towards the end of summer, so feel free to take over this if you're interested in the meantime.

Not holding my breath, but also hoping that dask.array support improves in the meantime as well. 

(The recent Array API updates suggest that sort is the most pressing thing that's missing in dask. Linalg wise, the eig family of methods is probably the next biggest missing feature in dask, we just haven't seen it come up yet since not a lot of estimators have been ported yet).